### ISSUE TITLE
"--config-file" option doesn't behave as expected when passing in home dir shortcut

### ISSUE THREAD

**Issue by Mierdin** [StackStorm/st2#3186]
The [CLI documentation](https://docs.stackstorm.com/reference/cli.html#configuration-file) gives an example of being able to pass a different configuration file into the `--config-file` argument, but this appears to just append this string to the current directory (implying a relative path is required here), rather than evaluate the provided argument as an absolute path:

```
(virtualenv)vagrant@st2dev:~/st2/st2client$ st2 --config-file=~/.st2/prod-config action list
Traceback (most recent call last):
  File "/home/vagrant/st2/virtualenv/bin/st2", line 11, in <module>
    load_entry_point('st2client==2.2.dev0', 'console_scripts', 'st2')()
  File "/home/vagrant/st2/st2client/st2client/shell.py", line 331, in main
    return Shell().run(argv)
  File "/home/vagrant/st2/st2client/st2client/shell.py", line 272, in run
    config = self._parse_config_file(args=args)
  File "/home/vagrant/st2/st2client/st2client/base.py", line 160, in _parse_config_file
    config_file_path = self._get_config_file_path(args=args)
  File "/home/vagrant/st2/st2client/st2client/base.py", line 179, in _get_config_file_path
    raise ValueError('Config "%s" not found' % (path))
ValueError: Config "/home/vagrant/st2/st2client/~/.st2/prod-config" not found
```

**Comment by Mierdin** [comment]
Indeed, this seems to work:

```
st2 --config-file=../../.st2/prod-config action list
```

I would prefer that this uses the absolute path, or if this was done by design, then the help output and st2 docs should be updated to be clearer here.

**Comment by Kami** [comment]
@Mierdin Yeah, I'm fine with calling `os.path.expanduser` on the path provided by the user via command line argument.

I believe doing that should make the example you pasted work (we might also need to call abspath on it, but not 100% sure).

**Comment by Kami** [comment]
Also, good catch - it seems like `os.path.abspath` call was added at some point later when docs were added so it broke.

Will make sure we have tests for it going forward :)

**Comment by lakshmi-kannan** [comment]
This seems fixed now. Closing. cc: @Kami

**Comment by Kami** [comment]
Yep, fixed in #3192. Thanks for closing.
